### DISCUSSION TITLE
how to render large number lines

### DISCUSSION THREAD

**Discussion by Tony-qty** [foxglove/community#900] [General]
When faced with the task of rendering 1296 LinePrimitiveRenderable instances, performance can significantly degrade.how to optimize this

**Comment by jtbandes** [comment]
Combine them into a single LinePrimitive with `type = LINE_LIST` for better performance. If many of the vertices are common to several line segments, use `indices` to avoid needing to duplicate data in the `points` array.

**Reply by Tony-qty** [reply]
thanks
